src/azul/service/manifest_service.py

```diff
@@ -890,23 +889,35 @@ def write_to(self, output: IO[str]) -> Optional[str]:
             for bundle in list(doc['bundles'])[0:100]:  # iterate over copy …
                 doc['bundles'] = [bundle]  # … to facilitate this in-place modification
                 row = {}
+                related_rows = []
                 for doc_path, column_mapping in self.manifest_config.items():
                     entities = self._get_entities(doc_path, doc)
                     self._extract_fields(entities, column_mapping, row)
+                    if doc_path == 'contents.files':
+                        entity = one(entities)
+                        try:
+                            related_files = entity['related_files']
+                        except KeyError:
+                            # Disregard file entities without related_files
+                            continue
+                        else:
+                            for file in related_files:
+                                related_row = {}
+                                invalid_manifest_entry = file.keys() - entity.keys()
+                                for field, value in file.items():
+                                    if field in invalid_manifest_entry:
+                                        continue
+                                    else:
+                                        entity[field] = value
```

Comment:
> tbh im not even sure this would make a performance impact, it seems like the less manipulation of objects the better.

Agreed. I was able to replace the whole section by calling on …
```
entity.update(file)
```
… so entity entries that are present in related `file` will be overwritten in one call. However, I removed the `indexed` field of the `related_file` object. For one, it isn't represented in the compact manifest, so it would not be a disadvantage or a potential problem. Also, I do not think it is significant for anything other than indicating wether the content of the document was indexed, which we know for related files, they do not get indexed, so we don't loose or gain information.

Reply:
entity.update() certainly makes the code more compact and legible. The downside to the approach is that we don't get the flexibility of being able to handle fields in the related file that we don't need in the file entity, also the data overwriting is still mostly occurring. 

> For one, it isn't represented in the compact manifest, so it would not be a disadvantage or a potential problem.
 
The PR is focused on compact manifests, but some of the changes are going to land in the indexer transformations. IIRC, by removing the `_indexed` field from the `_related_files` we also loose that field within the related_files object during a file hits response from ES.

It probably something we can live with, but if the field was added in the future would this cause problems with the compact manifest generation? Would there be an extra `_indexed` column in the manifest row? 

I'm not sure how high these concerns are, it might be something you want @hannes-ucsc to chime in on, lots of trade offs happening.